pkg/resource/unstructured/composite/composite.go

```diff
@@ -273,3 +281,32 @@ func (c *Unstructured) GetObservedGeneration() int64 {
 	_ = fieldpath.Pave(c.Object).GetValueInto("status", status)
 	return status.GetObservedGeneration()
 }
+
+// SetClaimConditions of this Composite resource.
+func (c *Unstructured) SetClaimConditions(conditionTypes ...xpv1.ConditionType) {
+	existingConditionSlice := c.GetClaimConditions()
+
+	existingConditionMap := make(map[xpv1.ConditionType]struct{}, len(existingConditionSlice))
+	for _, cnd := range existingConditionSlice {
+		existingConditionMap[cnd] = struct{}{}
+	}
+
+	for _, ct := range conditionTypes {
+		if xpv1.IsSystemConditionType(ct) {
+			continue
+		}
+		if _, ok := existingConditionMap[ct]; ok {
+			continue
+		}
+		existingConditionMap[ct] = struct{}{}
+		existingConditionSlice = append(existingConditionSlice, ct)
+	}
+	_ = fieldpath.Pave(c.Object).SetValue("status.claimConditions", existingConditionSlice)
+}
+
+// GetClaimConditions of this Composite resource.
+func (c *Unstructured) GetClaimConditions() []xpv1.ConditionType {
```

Comment:
Good point.
I like changing it to `GetClaimConditionTypes`, but it would no longer match the API object. Can we also change the field on the API object to `claimConditionTypes` since your point about `conditions.(type) != claimConditions.(type)` applies to the API object as well?

Reply:
> Can we also change the field on the API object to claimConditionTypes since your point about conditions.(type) != claimConditions.(type) applies to the API object as well?

For some reason it seems less of an issue there to me, but yeah I think all things considered it makes sense to change it to `status.claimConditionTypes`.